### PR TITLE
Add support for strictBooleanEnforced

### PR DESCRIPTION
**Pull request by kibertoad**
#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,6 +103,9 @@ Default value is `false`
 `eagerInject` - whether to process `eagerInject` fields in DI resolver configuration, which instantiates and caches module immediately. Disabling this will make app startup slightly faster.
 Default value is `false`
 
+`strictBooleanEnforced` - whether to throw an error if `enabled` field in a resolver configuration is set with unsupported value (anything different from `true` and `false`). It is recommended to set this to `true`, which will be a default value in the next semver major release.
+Default value is `false`
```
Comment (Eomm) [inline README.md:107]: I was thinking to set it to true already, but let's sync with the wrapped module tho

https://www.npmjs.com/package/awilix-manager

**Review by Eomm — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (kibertoad) [reply README.md:107]: @Eomm I'm OK to set it to true and release a new semver major.

Comment (kibertoad) [reply README.md:107]: although we can do it in a follow-up PR for backwards compatibility

Comment (Eomm) [reply README.md:107]: No, it is not mentioned as security warning - so it is good this default 👍🏼

**Pull request merged**